app/src/main/java/edu/csumb/caitlin/lo/cst438_s21_proj01_group10/db/AppDAO.java

```diff
@@ -24,4 +24,19 @@
     /* Get user by ID */
     @Query("SELECT * FROM " + AppDatabase.USER_TABLE + " WHERE userId LIKE :userId")
     User getUserById(int userId);
+
+
+    /* Favorites Queries */
+    @Insert
+    void insert(Favorites... favorites);
+
+    @Update
+    void update(Favorites... favorites);
+
+    @Delete
+    void delete(Favorites favorites);
+
```

Comment:
Maybe add a @Query to get database of user login info: kinda like ("SELECT * FROM " + AppDatabase.USER_TABLE + " WHERE username = :username and password = :password")
    User getUser(String username, String password);

Reply:
using the getUserByUsername will get all of the user information (userId, username, and password) in a User object, which you can then use to compare with whatever they entered into the password field.

if we select where username and password match, we can't have two separate checks for username and password and you can only say "no such user exists", even if they had correct username but wrong password